src/Shared.CLI/Helpers/PackageDownloader.cs

```diff
@@ -248,7 +248,7 @@ public async Task<List<PackageURL>> GetPackageVersionsToProcess(PackageURL purl)
                 {
                     try
                     {
-                        foreach (string? version in await packageManager.EnumerateVersions(purl, doCache))
+                        foreach (string? version in await packageManager.EnumerateVersionsAsync(purl, doCache))
```

Comment:
```suggestion
                        await foreach (string? version in packageManager.EnumerateVersionsAsync(purl, doCache))
```

Reply:
Does this actually produce nulls in the enumeration?

Reply:
I don't think I can do this because `EnumerateVersionsAsync` returns a `Task<IEnumberable<string>>` not a `IAsyncEnumerable<string>`

Reply:
![image](https://user-images.githubusercontent.com/8357054/163276297-91396f20-10d4-4ab4-a00e-aecad2df07a4.png)

Reply:
Ah I see. Other way is fine then.